### PR TITLE
feat: support restore window decorations

### PR DESCRIPTION
**Pull request by mantou132**
when user use `window.set_decorations(false)` in `builder.setup`, restore window size is incorrect

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ use std::{
   sync::{Arc, Mutex},
 };
 
-const STATE_FILENAME: &str = ".window-state";
+pub const STATE_FILENAME: &str = ".window-state";
```
Comment (amrbashir) [inline src/lib.rs:18]: why does this need to be public?

File: src/lib.rs [hunk 2]
```diff
@@ -93,6 +95,7 @@ impl<R: Runtime> WindowExt for Window<R> {
       let PhysicalPosition { x, y } = self.outer_position()?;
       let maximized = self.is_maximized().unwrap_or(false);
       let visible = self.is_visible().unwrap_or(true);
+      let decorated = self.is_decorated().unwrap_or(true);
```
Comment (amrbashir) [inline src/lib.rs:98]:
```suggestion
      let decorated = self.is_decorated().unwrap_or_else(|| true);
```

File: src/lib.rs [hunk 3]
```diff
@@ -178,6 +182,7 @@ impl Builder {
             let mut c = cache.lock().unwrap();
             if let Some(state) = c.get_mut(&label) {
               let is_maximized = window_clone.is_maximized().unwrap_or(false);
+              state.decorated = window_clone.is_decorated().unwrap_or(true);
```
Comment (amrbashir) [inline src/lib.rs:185]:
```suggestion
              state.decorated = window_clone.is_decorated().unwrap_or_else(|| true);
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (mantou132) [reply src/lib.rs:18]: This is a temporary solution that allows users to delete state.

Comment (amrbashir) [reply src/lib.rs:18]: I guess it is fine to expose it.

File: src/lib.rs (hunk 2 above)
Comment (mantou132) [reply src/lib.rs:98]: has been changed. but why?

Comment (amrbashir) [reply src/lib.rs:98]: explained here https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or

Comment (amrbashir) [reply src/lib.rs:98]: nvm, looks like clippy is happy with `unwrap_or`

**Review by amrbashir — APPROVED** [review]

**Pull request merged**